packages/plugins/src/hardhat/simulate.ts

```diff
@@ -145,10 +148,24 @@ export const simulate = async (
   }
 
   if ((await isLiveNetwork(provider)) || (await isFork(provider))) {
-    // Use the same block number as the Forge script that collected the user's transactions. This
-    // reduces the chance that the simulation throws an error or stalls, which can occur when using
-    // the most recent block number.
-    envVars['SPHINX_INTERNAL__BLOCK_NUMBER'] = networkConfig.blockNumber
+    // Use the block number from the Forge script minus the largest possible chain reorg size, which
+    // is determined by Hardhat. We must subtract the reorg size so that Hardhat caches the RPC
+    // calls in the simulation. Otherwise, Hardhat will send hundreds of RPC calls, which frequently
+    // causes rate limit errors, especially for public or free tier RPC endpoints.
+    //
+    // Subtracting the reorg size can lead to the following edge case:
+    // 1. User executes a transaction on the live network.
+    // 2. User calls Sphinx's Propose or Deploy command using a script that relies on the state that
+    //    resulted from the transaction in the previous step.
+    // 3. The collection process works correctly because Foundry uses the latest block number.
+    // 4. The simulation uses a block where the transaction doesn't exist yet, causing an error.
+    //
+    // This edge case will naturally resolve itself if the user continues to attempt to
+    // propose/deploy. This is because the corresponding block will eventually be included in the
+    // simulation after there have been enough block confirmations.
```

Comment:
Made a [ticket](https://linear.app/chugsplash/issue/CHU-837/resolve-edge-case-caused-by-using-an-older-block-number-in-simulation) to resolve this edge case. I think this situation is unlikely to happen in practice. The reorg size is pretty modest, e.g. 5 blocks on Ethereum, and 30 blocks on most other networks. I think a reorg size of 30 corresponds to 15 minutes on Rootstock, which perhaps the slowest network we support